wheel/src/api.rs

```diff
@@ -495,6 +498,164 @@ pub fn py_calculate_ip_iters(
+    let refs = block_refs
+        .into_iter()
+        .map(|b| {
+            let buf = b
+                .extract::<PyBuffer<u8>>()
+                .expect("block_refs must be list of buffers");
+            py_to_slice::<'a>(buf)
+        })
+        .collect::<Vec<&'a [u8]>>();
+    let args = setup_generator_args(&mut a, refs)?;
+    let dialect = ChiaDialect::new(flags);
+
+    let Reduction(_clvm_cost, res) = run_program(
+        &mut a,
+        &dialect,
+        program,
+        args,
+        constants.max_block_cost_clvm,
+    )
+    .map_err(|_| ValidationErr(program, ErrorCode::GeneratorRuntimeError))?;
+
+    let (first, _rest) = a
+        .next(res)
+        .ok_or(ValidationErr(res, ErrorCode::GeneratorRuntimeError))?;
+    let mut cache = TreeCache::default();
+    let mut iter = first;
+    while let Some((spend, rest)) = a.next(iter) {
+        iter = rest;
+        let [_, puzzle, _] = extract_n::<3>(&a, spend, ErrorCode::InvalidCondition)?;
+        cache.visit_tree(&a, puzzle);
+    }
+    iter = first;
+    while let Some((spend, rest)) = a.next(iter) {
+        iter = rest;
+        let mut cond_output = Vec::<(u32, Vec<Py<PyBytes>>)>::new();
+        let Ok([parent_id, puzzle, amount, solution, _spend_level_extra]) =
+            extract_n::<5>(&a, spend, ErrorCode::InvalidCondition)
+        else {
+            continue; // if we fail at this step then maybe the generator was malicious - try other spends
+        };
+        let puzhash = tree_hash_cached(&a, puzzle, &mut cache);
+        let parent_id = BytesImpl::<32>::from_clvm(&a, parent_id)
+            .map_err(|_| ValidationErr(first, ErrorCode::InvalidParentId))?;
+        let coin = Coin::new(
+            parent_id,
+            puzhash.into(),
+            u64::from_clvm(&a, amount)
+                .map_err(|_| ValidationErr(first, ErrorCode::InvalidCoinAmount))?,
+        );
+        let Ok(puzzle_program) = Program::from_clvm(&a, puzzle) else {
+            continue;
+        };
+        let Ok(solution_program) = Program::from_clvm(&a, solution) else {
+            continue;
+        };
+        let coinspend = CoinSpend::new(coin, puzzle_program.clone(), solution_program.clone());
+        let Ok((_, res)) =
+            puzzle_program.run(&mut a, flags, constants.max_block_cost_clvm, &solution)
+        else {
+            continue; // Skip this spend on error
+        };
+        // conditions_list is the full returned output of puzzle ran with solution
+        // ((51 0xcafef00d 100) (51 0x1234 200) ...)
+
+        // condition is each grouped list
+        // (51 0xcafef00d 100)
+        let mut num = 0;
+        let mut iter_two = res;
+        while let Some((condition, rest_two)) = a.next(iter_two) {
+            iter_two = rest_two;
+            let mut iter_three = condition;
+            let mut count: usize = 0;
+            let mut bytes_vec = Vec::<Py<PyBytes>>::new();
```

Comment:
not a big deal, but I think you could remove `count` and just use `bytes_vec.len()` instead. setting `count = 0` would have to be `bytes_vec.clear()` instead.

Reply:
ended up using `num` as the flag instead because of `REMARK` as a condition, but I have removed `count` now